internal/repository/prisma/dbsqlc/step_runs.sql

```diff
@@ -306,18 +306,24 @@ WITH valid_workers AS (
     SELECT
         DISTINCT ON (w."id")
         w."id",
-        COALESCE(SUM(ws."slots"), 100) AS "slots"
+        slots."remainingSlots" AS "slots"
     FROM
         "Worker" w
-    LEFT JOIN
-        "WorkerSemaphore" ws ON ws."workerId" = w."id"
+  LEFT JOIN (
+    SELECT wss."workerId", (COALESCE(w."maxRuns", 100) - COUNT(wss."id")) AS "remainingSlots"
```

Comment:
This inner select is joining all worker semaphores with workers in the entire table, which is going to get very slow. Why not just return the worker ids from `valid_workers` in the first CTE and use the result to get all worker semaphores which match condition `"workerId" IN (valid_workers)`?

Reply:
I noticed we're doing something similar below in the `Acquire` query:

```sql
    -- (in a second CTE)
    SELECT wss."id" AS "slotId", wss."workerId" AS "workerId"
    FROM "WorkerSemaphoreSlot" wss
    JOIN valid_workers w ON wss."workerId" = w."id"
    WHERE wss."stepRunId" IS NULL
```

Reply:
good catch